indico/modules/events/registration/client/js/components/ConsentToPublishEditor.jsx

```diff
@@ -0,0 +1,69 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2022 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import registrationChangeConsent from 'indico-url:event_registration.api_registration_change_consent';
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {Icon} from 'semantic-ui-react';
+
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+
+import ConsentToPublishDropdown, {publishModePropType} from './ConsentToPublishDropdown';
+
+import './ConsentToPublishEditor.module.scss';
+
+export default function ConsentToPublishEditor({
+  locator,
+  publishToParticipants,
+  publishToPublic,
+  initialConsentToPublish,
+}) {
+  const [consentToPublish, setConsentToPublish] = useState(initialConsentToPublish);
+  const [loading, setLoading] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const [failure, setFailure] = useState(false);
+
+  async function editConsentToPublish(value) {
+    const oldValue = consentToPublish;
+    setConsentToPublish(value);
+    setSuccess(false);
+    setFailure(false);
+    setLoading(true);
+    try {
+      await indicoAxios.post(registrationChangeConsent(locator), {consent_to_publish: value});
+      setLoading(false);
+      setSuccess(true);
+    } catch (error) {
+      setLoading(false);
+      setFailure(true);
+      setConsentToPublish(oldValue);
+      return handleAxiosError(error);
+    }
+  }
+
+  return (
+    <div styleName="consent-dropdown-container">
+      <ConsentToPublishDropdown
+        publishToParticipants={publishToParticipants}
+        publishToPublic={publishToPublic}
+        onChange={(evt, {value}) => editConsentToPublish(value)}
+        value={consentToPublish}
+        loading={loading}
+      />
+      {success && <Icon name="check" color="green" />}
+      {failure && <Icon name="x" color="red" />}
+    </div>
+  );
+}
+
+ConsentToPublishEditor.propTypes = {
+  locator: PropTypes.object.isRequired,
+  publishToParticipants: publishModePropType.isRequired,
+  publishToPublic: publishModePropType.isRequired,
+  initialConsentToPublish: PropTypes.oneOf(['not_given', 'participants', 'all']).isRequired,
```

Comment:
I'd use 'nobody' instead of 'not_given' here. I think that fits nicely with the other ones - "show to nobody / participants / all"